### ISSUE TITLE
[BITV] Share tab a11y issues

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#33740]
Share tab accessibility issues:

- https://github.com/nextcloud/nextcloud-vue/pull/3435
    - [x] Search suggestions are not read out by screenreader, multiselect component is not accesssible at all
    - [x] Subline of search suggestions needs more contrast, needs to use `color-text-maxcontrast`
    - [x] "No recommendations. Start typing." also needs to use `color-text-maxcontrast` (`#no-options` vue-select slot uses `--color-main-text`)

---

* [x] Search suggestions: https://github.com/nextcloud/nextcloud-vue/issues/2504
  * [x] suggestion should be button instead of link as it doesn’t load a new page (already not link)
* [x] Share link flow
  * [x] Currently share link button only says "Button", needs individual label as there can be multiple share links (correct label "Create a new share link" present on `master`)
  * [x] Focus feedback of menu goes full width
  * [x] As there may be multiple share links each one should have a unique label https://github.com/nextcloud/server/pull/36056
  	* [x] https://github.com/nextcloud/server/pull/36129

**Comment by Pytal** [comment]
* [ ] "Copy to clipboard" should be button instead of link https://github.com/nextcloud/server/pull/35240

**Comment by Pytal** [comment]
- [ ] All "save" buttons need to be buttons instead of links, be keyboard-focusable, and have an accessible label, and showSuccess notification https://github.com/nextcloud/server/pull/35271
  - [x] Save share label
  - [x] Password save
  - [x] Share note
  - [x] Unshare

**Comment by Pytal** [comment]
- [x] Avatar has label "Avatar of displayname"

Done in https://github.com/nextcloud/nextcloud-vue/blob/556642f23177563da62afdfc798837380e82c04e/src/components/NcAvatar/NcAvatar.vue#L373-L382

**Comment by Pytal** [comment]
- [ ] Search field needs a permanent label https://github.com/nextcloud/server/pull/35272

**Comment by Pytal** [comment]
Transferred ""Share label" needs visible text label above"
- https://github.com/nextcloud/server/issues/35416

**Comment by Pytal** [comment]
Transferred ""Link copied" needs to use showSuccess which includes aria-live"
- https://github.com/nextcloud/server/issues/35438

**Comment by Pytal** [comment]
Transferred "Entries use transparency when not active, leading to contrast issues"
- https://github.com/nextcloud/nextcloud-vue/issues/3521

**Comment by Pytal** [comment]
Transferred "When leaving the share link menu, one is being put back into the file list instead of just closing the menu"
- https://github.com/nextcloud/server/issues/35531

**Comment by Pytal** [comment]
Transferred into https://github.com/nextcloud/nextcloud-vue/issues/3526
* Focus of search suggestions is not clearly visible, needs border instead of background
* aria-label needs to be removed for unclickable avatars

**Comment by Pytal** [comment]
> Focus feedback of menu goes full width

Might be missing something but all the focus feedback in the share tab seem to be normal, is there a specific menu that needs to be changed @PVince81?

**Comment by Pytal** [comment]
> Share link row should be fully focusable / clickable to open menu.

![image](https://user-images.githubusercontent.com/24800714/210678789-f0cce99f-985c-4bb9-98c7-696901682334.png)

Does not seem necessary but if it is then we may have to redesign all the rows. Does the current design with only the highlighted button area itself being activatable work accessibility-wise @michaelnissenbaum? Or would the whole row have to be activatable?

**Comment by Pytal** [comment]
> @Pytal this is a rather old ticket so it's likely that some issues I mentioned were already fixed recently

Seems fixed after investigations

**Comment by Pytal** [comment]
Transferred "Share link row should be fully focusable / clickable to open menu."
- #36111
